EngineLayer/ClassicSearch/ClassicSearchEngine.cs

```diff
@@ -167,6 +184,34 @@ protected override MetaMorpheusEngineResults RunSpecific()
             return new MetaMorpheusEngineResults(this);
         }
 
+        private void AddPeptideCandidateToPsm(ScanWithIndexAndNotchInfo scan, object[] myLocks, double thisScore, PeptideWithSetModifications peptide, List<MatchedFragmentIon> matchedIons)
+        {
+            bool meetsScoreCutoff = thisScore >= CommonParameters.ScoreCutoff;
+
+            // this is thread-safe because even if the score improves from another thread writing to this PSM,
+            // the lock combined with AddOrReplace method will ensure thread safety
+            if (meetsScoreCutoff)
```

Comment:
Can we keep this conditional outside of the method?

Reply:
We can do it but if keep it outside, then we will have to add one line to calculate the "meetsScoreCutOff" every time we call this method.